Add Docments to tabular.data

Karthik-d-k opened the pull request:
This is a simple PR to add Docments to notebook -> 41_tabular.data. I tried my best to align with the style guide provided.
Happy to do some modifications if there are any corrections.


Notes :-
- For input `rm_type_tfms` in `test_dl` method -> I'm not completely sure if i did justification, my understanding might be wrong here though, looking forward to learn things in my first PR towards open source 😄

kevinbird15 commented on fastai/tabular/data.py:
```diff
@@ -12,22 +12,39 @@ class TabularDataLoaders(DataLoaders):
-                valid_idx=None, **kwargs):
-        "Create from `df` in `path` using `procs`"
+    def from_df(cls,
+                df:pd.DataFrame, # A Dataframe object
```
```suggestion
                df:pd.DataFrame, 
```
No need to point out it is a dataframe object since it is named df and has type pd.DataFrame

kevinbird15 commented on fastai/tabular/data.py:
```diff
@@ -12,22 +12,39 @@ class TabularDataLoaders(DataLoaders):
+                cont_names:list=None, # Column names pertaining to Continuous variables
+                y_names:list=None, # Names of the Dependent variables
+                y_block:TransformBlock=None, # `TransformBlock` to use for the Target
+                valid_idx:list=None, # List of indices to use for the Validation set (defaults to a random split otherwise)
```
I like this defaults comment since otherwise all we see is default is None.  Nice work :)

Karthik-d-k replied:
Thanks @kevinbird15 :)

kevinbird15 commented on fastai/tabular/data.py:
```diff
@@ -12,22 +12,39 @@ class TabularDataLoaders(DataLoaders):
+                y_block:TransformBlock=None, # `TransformBlock` to use for the Target
+                valid_idx:list=None, # List of indices to use for the Validation set (defaults to a random split otherwise)
+                **kwargs):
+        "Create `DataLoaders` from `df` in `path` using `procs`"
```
```suggestion
        "Create `TabularDataLoaders` from `df` in `path` using a list of `TabularProc`s"
```
1. I think this actually creates a TabularDataLoaders
2. Thoughts on passing TabularProc here instead of procs so that it links to that section of docs?

Karthik-d-k replied:
Perfect, Accepted 👍

kevinbird15 commented on fastai/tabular/data.py:
```diff
@@ -12,22 +12,39 @@ class TabularDataLoaders(DataLoaders):
-    def from_csv(cls, csv, skipinitialspace=True, **kwargs):
-        "Create from `csv` file in `path` using `procs`"
+    def from_csv(cls,
+                 csv:str, # Location of a `csv` file
```
```suggestion
                 csv:(str,Path,io.BufferedReader), # `csv` file to be ingested
```
Not sure I love my wording here.  Just wanted to mostly show that it can also handle Path and even a BufferedReader since all this does is process pd.read_csv

Karthik-d-k replied:
That makes sense, 
Is it fine to take over the type-hint's and comments directly from pandas for this, which is as follows -->
 ```suggestion
                 csv:FilePathOrBuffer, # str, path object or file-like object
```
which satisfies all the required suggestions 😉

Karthik-d-k replied:
If i use `FilePathOrBuffer`, it was throwing error, so i have used your suggestion and updated as follows ->
`csv:(str,Path,io.BufferedReader), # str, path object or file-like object`

kevinbird15 commented on fastai/tabular/data.py:
```diff
@@ -12,22 +12,39 @@ class TabularDataLoaders(DataLoaders):
+                test_items, # Items to create new test dataset based on training data
+                rm_type_tfms=None, # Number of `Transform`s to be removed from `procs`
+                process:bool=True, # Apply validation `TabularProc`s (non-lazy `Transform`s) on `test_items` if `True`
+                inplace:bool=False, # Keep separate copy of original `test_items` in memory if `True`
```
```suggestion
                inplace:bool=False, # Keep separate copy of original `test_items` in memory if `False`
```
I think this is False, but I will validate that if you thing it's opposite.  Let me know if you need me to dig deeper into this one

Karthik-d-k replied:
Sorry, my bad, it should be `false`

kevinbird15 commented on fastai/tabular/data.py:
```diff
@@ -12,22 +12,39 @@ class TabularDataLoaders(DataLoaders):
-    def test_dl(self, test_items, rm_type_tfms=None, process=True, inplace=False, **kwargs):
+    def test_dl(self,
+                test_items, # Items to create new test dataset based on training data
+                rm_type_tfms=None, # Number of `Transform`s to be removed from `procs`
```
I don't see where `rm_type_tfms` is actually being used in this test_dl.  @muellerzr This might be another one for us to discuss further about either removing or adding the logic. Here is the similar code from data.core: https://github.com/fastai/fastai/blob/63c359fcf2389659c58fe9caea0eb44a3432858b/fastai/data/core.py#L395-L402

Karthik-d-k replied:
Yes, i was very confused by this input and went into a loop hole to understand what is happening, but no luck.
I have also tested setting it to values 1/2/3 thinking it might remove the `TabularProc`s from being applied on test data, but that was not the case..!

Karthik-d-k commented on nbs/41_tabular.data.ipynb:
```diff
@@ -581,6 +581,32 @@
+    }
+   ],
+   "source": [
+    "show_doc(TabularDataLoaders.test_dl)"
```
Added document for `TabularDataLoaders.test_dl` as it was missing in base.

kevinbird15 replied:
We shouldn't need to do this, but I'm not sure why it isn't showing up when the docs are generated.  Still hoping somebody else has an idea on why this one is having issues generating

Karthik-d-k replied:
Actually, in the base NB,  `show_doc` is present for the following 2 methods before they have been used in example ->
- `show_doc(TabularDataLoaders.from_df)`
- `show_doc(TabularDataLoaders.from_csv)`

So i have added the following missing method so that it will be consistent (all 3 methods are present w/ show_doc now)->
- `show_doc(TabularDataLoaders.test_dl)`

Karthik-d-k commented on fastai/tabular/data.py:
```diff
@@ -13,16 +13,16 @@ class TabularDataLoaders(DataLoaders):
                 valid_idx:list=None, # List of indices to use for the Validation set (defaults to a random split otherwise)
                 **kwargs):
-        "Create `TabularDataLoaders` from `df` in `path` using a list of `TabularProc`s"
+        "Create `TabularDataLoaders` from `df` in `path` using `procs`"
```
reverted back to `procs` as i was repeating info as per suggestion from @muellerzr during live stream w/ w&b 😄

kevinbird15 replied:
I like that.  Good change

Karthik-d-k commented on fastai/tabular/data.py:
```diff
@@ -12,22 +12,39 @@ class TabularDataLoaders(DataLoaders):
 
     @delegates(TabDataLoader.__init__)
-    def test_dl(self, test_items, rm_type_tfms=None, process=True, inplace=False, **kwargs):
+    def test_dl(self,
```
`show_doc` for `test_dl` was missing in base NB, now i have added it in my latest commit

muellerzr commented on fastai/tabular/data.py:
```diff
@@ -12,22 +12,39 @@ class TabularDataLoaders(DataLoaders):
+                process:bool=True, # Apply validation `TabularProc`s (non-lazy `Transform`s) on `test_items` if `True`
+                inplace:bool=False, # Keep separate copy of original `test_items` in memory if `False`
+                **kwargs):
+        "Create Test `TabDataLoader` from `test_items` using validation `procs`"
```
```suggestion
        "Create test `TabDataLoader` from `test_items` using validation `procs`"
```

muellerzr reviewed (changes requested):
Thanks! Added some nits, please fix 😄

Karthik-d-k commented:
> Thanks! Added some nits, please fix 😄

Thank you very much 😃 
I have fixed all the suggestions/corrections provided by both of you (@muellerzr , @kevinbird15 )

Only thing I'm not sure about is the following comment -->
```
csv:(str,Path,io.BufferedReader), # A csv from which features/targets are inferred
```
Please let me know if i missed something, i will fix it 🚀

Karthik-d-k commented:
Accidentally closed the PR 😭

muellerzr commented on fastai/tabular/data.py:
```diff
@@ -12,22 +12,42 @@ class TabularDataLoaders(DataLoaders):
-    def from_csv(cls, csv, skipinitialspace=True, **kwargs):
-        "Create from `csv` file in `path` using `procs`"
+    def from_csv(cls,
+        csv:(str,Path,io.BufferedReader), # A csv from which features/targets are inferred
```
This isn't accurate. 

```suggestion
        csv:(str,Path,io.BufferedReader), # A csv of training data
```

Karthik-d-k replied:
Done ✨

muellerzr reviewed (changes requested):
One nit

muellerzr commented:
Great work!

Karthik-d-k commented:
> Great work!

Thank you very much for the opportunity, and also thanks for all the support without which i wouldn't  be able to contribute my first open source contribution 😄 🚀 
Looking forward for many more contributions in future..!
